src/Qir/Execution/QirCommandLineTool/Program.cs

```diff
@@ -0,0 +1,156 @@
+            {
+                Required = true
+            };
+            buildCommand.AddOption(qsharpDllOption);
+
+            Option<DirectoryInfo[]> libraryDirectories = new Option<DirectoryInfo[]>(
+                aliases: new string[] { "--libraryDirectories", "--lib" },
+                description: "One or more paths to the directories containing the libraries to be linked.")
+            {
+                Required = true
+            };
+            buildCommand.AddOption(libraryDirectories);
+
+            Option<DirectoryInfo[]> includeDirectories = new Option<DirectoryInfo[]>(
+                aliases: new string[] { "--includeDirectories", "--include" },
+                description: "The path to the directory containing the headers required for compilation.")
+            {
+                Required = true
+            };
+            buildCommand.AddOption(includeDirectories);
+
+            Option<DirectoryInfo> executablesDirectory = new Option<DirectoryInfo>(
+                aliases: new string[] { "--executablesDirectory", "--exe" },
+                description: "The path to the output directory where the created executables will be placed.")
+            {
+                Required = true
+            };
+            buildCommand.AddOption(executablesDirectory);
+
+            return buildCommand;
+        }
+
+        /// <summary>
+        /// Copies the handle and options from the given sub command to the given command.
+        /// </summary>
+        /// <param name="root">The command whose handle and options will be set.</param>
+        /// <param name="subCommand">The sub command that will be copied from.</param>
+        private static void SetSubCommandAsDefault(Command root, Command subCommand)
+        {
+            root.Handler = subCommand.Handler;
+            foreach (var option in subCommand.Options)
+            {
+                root.AddOption(option);
+            }
+        }
+
+        /// <summary>
+        /// A modification of the command-line <see cref="HelpBuilder"/> class.
+        /// </summary>
+        private sealed class QsHelpBuilder : HelpBuilder
+        {
+            /// <summary>
+            /// Creates a new help builder using the given console.
+            /// </summary>
+            /// <param name="console">The console to use.</param>
+            internal QsHelpBuilder(IConsole console) : base(console)
+            {
+            }
+
+            protected override string ArgumentDescriptor(IArgument argument)
+            {
+                // Hide long argument descriptors.
+                var descriptor = base.ArgumentDescriptor(argument);
+                return descriptor.Length > 30 ? argument.Name : descriptor;
+            }
+        }
+
+        /// <summary>
+        /// A class for encapsulating the different options for the build command.
+        /// </summary>
+        public sealed class BuildOptions
```

Comment:
This looks like an ideal case for a record class (I think .NET 5 is used to build?). To use record classes in projects compatible with .NET Core SDK 3.1, you just need to make sure that the following declaration occurs somewhere in your project:

```
namespace System.Runtime.CompilerServices
{
    public class IsExternalInit { }
}
```

Reply:
I'm not very excited to use the new C# record types, and I don't really see the advantage here. Using a public class like this is consistent with examples and tutorials of the System.CommandLine Commands, and the compatibility with the new record feature is questionable. We have something that works well here.